File: src/php/reports/create-reports.php
<?php

require_once("/Users/jhammond/git/easypost/easypost-tools/vendor/autoload.php");
use \EasyPost\EasyPost;
use \Dotenv\Dotenv;

// API Key
$dotenv = Dotenv::createImmutable("/Users/jhammond/git/easypost/easypost-tools");
$dotenv->load();
EasyPost::setApiKey(getenv("EASYPOST_PROD_API_KEY"));

// Set a random date to use
$year = sprintf('%04d', rand(0001,2020));
$month = sprintf('%02d', rand(01,12));

$payment_log_report = \EasyPost\Report::create(array(
  "start_date" => "$year-$month-01",
  "end_date" => "$year-$month-31",
  "type" => "payment_log"
));
  
$refund_report = \EasyPost\Report::create(array(
  "start_date" => "$year-$month-01",
  "end_date" => "$year-$month-31",
  "type" => "refund"
));

$shipment_report = \EasyPost\Report::create(array(
  "start_date" => "$year-$month-01",
  "end_date" => "$year-$month-31",
  "type" => "shipment"
));

$tracker_report = \EasyPost\Report::create(array(
  "start_date" => "$year-$month-01",
  "end_date" => "$year-$month-31",
  "type" => "tracker"
));

echo "$payment_log_report \n $refund_report \n $shipment_report \n $tracker_report";

File: src/python/reports/create_reports.py
import os

import easypost
from dotenv import load_dotenv

load_dotenv()
easypost.api_key = os.getenv('EASYPOST_PROD_API_KEY')

payment_log_report = easypost.Report.create(
    start_date='2019-08-01',
    end_date='2019-08-23',
    type='payment_log'
)

refund_report = easypost.Report.create(
    start_date='2019-08-01',
    end_date='2019-08-23',
    type='refund'
)

shipment_report = easypost.Report.create(
    start_date='2019-08-01',
    end_date='2019-08-23',
    type='shipment'
)

tracker_report = easypost.Report.create(
    start_date='2019-08-01',
    end_date='2019-08-23',
    type='tracker'
)

print(payment_log_report)
print(refund_report)
print(shipment_report)
print(tracker_report)
